File: HLC01_03Java/src/main/java/org/example/gestorTareas05/Proyect.java
package org.example.gestorTareas05;

import java.util.ArrayList;

class Proyect {
    private ArrayList<Task> tasks;
    private String name;

    public Proyect(String name) {
        this.name = name;
        this.tasks = new ArrayList<>();
    }


    public String getName() {
        return this.name;
    }


    public void addTask(Task task) {
        tasks.add(task);
        System.out.println("Tarea añadida a " + this.name);
    }


    @Override
    public String toString() {
        StringBuilder tasksString = new StringBuilder();
        tasks.forEach(task -> tasksString.append("\n    " + task.toString()));

        return "Proyecto " + this.name + tasksString;
    }

}

File: HLC01_03Kotlin/src/main/kotlin/gestorTareas05/Proyect.kt
package org.example.gestorTareas05

class Proyect(val name: String) {
    private val tasks = mutableListOf<Task>()


    fun addTask(task: Task) {
        tasks.add(task)
        println("Tarea añadida a $name")
    }

    override fun toString(): String {
        return "Proyecto: $name ${tasks.forEach { println("        $it") }}"
    }

}